### DISCUSSION TITLE
How to work with Addressable

### DISCUSSION THREAD

**Discussion by TrongHieu90** [Cysharp/UniTask#409] [Q&A] [answered]
```
 public GameObject CreateItem(string itemID)
    {
        return InstantiateIconID(itemID).GetAwaiter().GetResult();
    }
    
    private async UniTask<GameObject> InstantiateIconID(string itemID)
    {
        string key = $"Spell_{itemID}";
        return await Addressables.InstantiateAsync(key);
    }
```

The following code throws an error "InvalidOperationException: Not yet completed, UniTask only allow to use await." when I call `CreateItem `method.

How should I work with addressable in this case ?

**Comment by neuecc** [comment, marked as answer]
It is not related addressable.
async method must be async, can't convert to sync.
